src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,73 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final Pair[] elements;
+    private int size;
+
+    public StorageImpl() {
+        elements = new Pair[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == MAX_SIZE) {
+            throw new ArrayIndexOutOfBoundsException("Array already full");
+        }
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i] == null) {
+                elements[size] = new Pair<>(key, value);
+                size++;
+                break;
+            } else if (elements[i].getKey() == key
+                    || elements[i].getKey() != null && elements[i].getKey().equals(key)) {
+                elements[i].setValue(value);
+                break;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i] == null) {
+                return null;
+            } else if (elements[i].getKey() == null && key == null
+                    || elements[i].getKey() != null && elements[i].getKey().equals(key)) {
+                return (V)elements[i].getValue();
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    private static class Pair<K,V> {
+        private K key;
+        private V value;
+
+        private Pair(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        private K getKey() {
+            return key;
+        }
+
+        private void setKey(K key) {
+            this.key = key;
+        }
+
+        private V getValue() {
+            return value;
+        }
+
+        private void setValue(V value) {
+            this.value = value;
+        }
```

Comment:
you did not fixed prev mentors comment
![image](https://github.com/mate-academy/jv-storage/assets/73041091/6c1b52b7-4ed1-42bf-8445-e9e22387ca4b)
```suggestion
        }
```